jetty-webapp/src/main/java/org/eclipse/jetty/webapp/CachingWebAppClassLoader.java

```diff
@@ -116,12 +151,45 @@ public URL getResource(String name)
     public void clearCache()
     {
         _cache.clear();
+        _classCache.clear();
         _notFound.clear();
     }
-    
+
+    public void setUseCache(boolean useCache)
+    {
+            this.useCache = useCache;
+    }
+
     @Override
     public String toString()
     {
         return "Caching["+super.toString()+"]";
     }
+
+    /**
+     * Lifecycle listener, that can be used from {@link WebAppContext#addLifeCycleListener(LifeCycle.Listener)}
+     * to switch off the {@link #_cache} and {@link #_classCache} after web application has been started.
+     */
+    public static class ClearCacheLifeCycleListener extends AbstractLifeCycle.AbstractLifeCycleListener
```

Comment:
Might want to make this `implements LifeCycle.Listener` instead.

Or at least implement `lifeCycleFailed(event)` and have the classloader cache cleaned up?